### PR TITLE
[document] Mention poetry.toml

### PR DESCRIPTION
**Pull request by majiang**
See https://github.com/python-poetry/poetry/issues/2937#issuecomment-709774431

# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ by passing the `--local` option to the `config` command.
 poetry config virtualenvs.create false --local
 ```
 
+Your local configuration of Poetry application is stored in `poetry.toml` file,
+which is separate from `pyproject.toml`.
+You should not check-in this file into your repository.
```
Comment (Secrus) [inline docs/configuration.md:35]: not sure if thats true. I would include this into repo.

Comment (majiang) [reply docs/configuration.md:35]: @finswimmer has written in https://github.com/python-poetry/poetry/issues/2937#issuecomment-696312341 :

> The `pyproject.toml` contains information about your package, like metadata and dependencies. Anything that is needed to build the package.
> 
> The `poetry.toml` contains the local configuration of the poetry application. One should not check-in this file into a repository, because these configuration are just personal flavors.

Should I use different wording if this is not *always* true?

Comment (Secrus) [reply docs/configuration.md:35]: oh, didn't know about that. Nevermind then.

**Comment by alexchandel** [comment]
Would be really, really convenient if you could put `poetry.toml` config values in `pyproject.toml`…

**Comment by majiang** [comment]
I was also a one who thought it is convenient, but after reading finswimmer's comment, the current behavior seems perfectly reasonable, except that this is not documented well.

File: docs/configuration.md (hunk 1 above)
Comment (abn) [inline docs/configuration.md:35]:
```suggestion
{{% note %}}
Your local configuration of Poetry application is stored in `poetry.toml` file,
which is separate from `pyproject.toml`.
You should not check-in this file into your repository.
{{% /note %}}
```

File: docs/configuration.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ by passing the `--local` option to the `config` command.
 poetry config virtualenvs.create false --local
 ```
 
+{{% note %}}
+Your local configuration of Poetry application is stored in `poetry.toml` file,
+which is separate from `pyproject.toml`.
+You should not check-in this file into your repository.
```
Comment (abn) [inline docs/configuration.md:36]: This is not entirely true or enforced. I'd reword it to suggest it is recommended not to commit it as it may contain user specific configuration or sensitive information.

**Comment by Secrus** [comment]
@majiang just a gentle bump ;) This would be nice addition to our docs.

File: docs/configuration.md (hunk 2 above)
Comment (majiang) [reply docs/configuration.md]: reworded!

**Comment by b-long** [comment]
There was no reply to my previous comment, so I'm unsure if anyone saw it.

I'd like to make one more plea for changing the wording.  Currently, the PR adds:

```
Your local configuration of Poetry application is stored in `poetry.toml` file,
which is separate from `pyproject.toml`.

It is recommended you do not check-in this file into your repository
since it may contain user specific or sensitive information.
```

When working in a shared code base, I find it extremely useful to have a `poetry.toml` alongside a `.vscode/`. That way, I can setup teammates quickly, and we all have the same configuration:

cd <project> && poetry install
code .
In the VSCode debugger, select "Python: Debug FastAPI"
Open your browser to http://localhost:8000/docs

I have a small FastAPI project which provides an example of this configuration: https://github.com/b-long/mooseblog .

As I said before, the `poetry.toml` file with configuration seems to be a popular feature. More people may adopt and love Poetry by learning about this feature (seeing it in open source repositories). Reading some previous discussions people seem to really like using it. For example: https://github.com/python-poetry/poetry/issues/108#issuecomment-584149674

**Comment by Secrus** [comment]
> There was no reply to my previous comment, so I'm unsure if anyone saw it.
> 
> I'd like to make one more plea for changing the wording. Currently, the PR adds:
> 
> ```
> Your local configuration of Poetry application is stored in `poetry.toml` file,
> which is separate from `pyproject.toml`.
> 
> It is recommended you do not check-in this file into your repository
> since it may contain user specific or sensitive information.
> ```
> 
> When working in a shared code base, I find it extremely useful to have a `poetry.toml` alongside a `.vscode/`. That way, I can setup teammates quickly, and we all have the same configuration...


Hi. We ask for rewording to avoid people showing up and claiming that we are recommending something not secure. `poetry.toml` may contain credentials to repositories for example. I will add my recommendation on how it could look.

File: docs/configuration.md [hunk 3]
```diff
@@ -30,6 +30,14 @@ by passing the `--local` option to the `config` command.
...
 ```
 
+{{% note %}}
+Your local configuration of Poetry application is stored in `poetry.toml` file,
+which is separate from `pyproject.toml`.
+
+It is recommended you do not check-in this file into your repository
+since it may contain user specific or sensitive information.
+{{% /note %}}
+
```
Comment (Secrus) [inline docs/configuration.md:40]:
```suggestion
{{% note %}}
Your local configuration of Poetry application is stored in `poetry.toml` file,
which is separate from `pyproject.toml`.
{{% /note %}}

{{% warning %}}
Be mindful about checking in this file into your repository since it may contain user-specific or sensitive information.
{{% /note %}}

```

**Comment by majiang** [comment]
committed @Secrus 's suggestion since it is better in that we clearly warn.

File: docs/configuration.md [hunk 4]
```diff
@@ -30,6 +30,15 @@ by passing the `--local` option to the `config` command.
 poetry config virtualenvs.create false --local
 ```
 
+{{% note %}}
+Your local configuration of Poetry application is stored in `poetry.toml` file,
```
Comment (neersighted) [inline docs/configuration.md:34]:
```suggestion
Your local configuration of Poetry application is stored in the `poetry.toml` file,
```

**Review by neersighted — CHANGES_REQUESTED** [review]

**Review by neersighted — APPROVED** [review]
